caliban_toolbox/figure_eight_functions_test.py

```diff
@@ -86,3 +96,105 @@ def test_create_job_urls():
+
+        # mock the call
+        response_dict = {'status_code': 200}
+        m.put(url, text=json.dumps(response_dict))
+        figure_eight_functions.upload_log_file(log_file=example_log_string, job_id=test_job_id,
+                                               key=test_key)
+
+# TODO: This mocks almost every call within the function, so we don't get proper integration
+# testing of the different parts working together. However, better than not testing at all
+
+
+@patch('caliban_toolbox.figure_eight_functions.upload_log_file')
+@patch('caliban_toolbox.figure_eight_functions.aws_upload_files')
+@patch('caliban_toolbox.figure_eight_functions.copy_job')
+@patch("getpass.getpass")
+def test_create_figure_eight_job(getpass, copy_job, aws_upload_files, upload_log_file):
+    getpass.return_value = 'test_api_key'
+    copy_job.return_value = 123
+    aws_upload_files.return_value = '200'
+    upload_log_file.return_value = 567
+
+    with tempfile.TemporaryDirectory() as temp_dir:
+
+        # create crop directory
+        crop_dir = os.path.join(temp_dir, 'crop_dir')
+        os.makedirs(crop_dir)
+        np.savez(os.path.join(crop_dir, 'test_crop.npz'))
+
+        figure_eight_functions.create_figure_eight_job(base_dir=temp_dir, job_id_to_copy=123,
+                                                       aws_folder='aws', stage='stage')
+
+
+def test_unzip_report():
+    with tempfile.TemporaryDirectory() as temp_dir:
+        # create example zip file
+        pathlib.Path(os.path.join(temp_dir, 'example_file.csv')).touch()
+        zip_path = os.path.join(temp_dir, 'job_report.zip')
+        zipfile.ZipFile(zip_path, mode='w').write(os.path.join(temp_dir, 'example_file.csv'))
+
+        figure_eight_functions.unzip_report(temp_dir)
+
+        assert os.path.exists(os.path.join(temp_dir, 'job_report.csv'))
+
+
+@patch('caliban_toolbox.figure_eight_functions.aws_download_files')
+@patch('caliban_toolbox.figure_eight_functions.download_report')
+def test_download_figure_eight_output(download_report, aws_download_files):
+
+    # we don't care about this return value, this is just to override existing function
+    download_report.return_value = 200
+    aws_download_files.return_value = 200
+
+    with tempfile.TemporaryDirectory() as temp_dir:
+
+        # create logs directory with zipped report
+        os.makedirs(os.path.join(temp_dir, 'logs'))
+        pathlib.Path(os.path.join(temp_dir, 'logs', 'example_file.csv')).touch()
+        zip_path = os.path.join(temp_dir, 'logs', 'job_report.zip')
+        zipfile.ZipFile(zip_path, mode='w').write(os.path.join(temp_dir, 'logs',
+                                                               'example_file.csv'))
+        # create log file
+        log_file = pd.DataFrame({'job_id': [1234]})
+        log_file.to_csv(os.path.join(temp_dir, 'logs', 'stage_0_upload_log.csv'))
+
+        figure_eight_functions.download_figure_eight_output(temp_dir)
```

Comment:
Can we make any correct-ness assertions? Right now the code does not fail, which is great, but is it actually doing the right thing?